### PR TITLE
Instrument the Azure API calls for Prometheus monitoring

### PR DESCRIPTION
**Pull request by ccojocar**
**What this PR does / why we need it**:

Instruments the Azure API calls for Prometheus monitoring.

**Special notes for your reviewer**:

This is version 2 based on the wrapped clients.

**Release note**:

```release-note
Instrument the Azure cloud provider for Prometheus monitoring.
```

cc @feiskyer @andyzhangx @jdumars @khenidak

### CODE REVIEW

**Review by feiskyer — COMMENTED** [review]
@cosmincojocar Seems all `ListNextResults` interfaces are not instrumented, why not also add them?

**Comment by cosmincojocar** [comment]
> Seems all ListNextResults interfaces are not instrumented, why not also add them?

That's right. It isn't obvious to me from where to read the `resource group` without modifying the interfaces. It doesn't appear to be available in the current object. 

One possibility is to add the resource group in the ListNextRresults method signatures. Do you have any other suggestions?

**Comment by feiskyer** [comment]
> One possibility is to add the resource group in the ListNextRresults method signatures. Do you have any other suggestions?

I thought resource group is included in `lastResult`, but it doesn't really after checking codes. I'm ok with add it in `ListNextResults()` signatures.

File: pkg/cloudprovider/providers/azure/azure_metrics.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```
Comment (andyzhangx) [inline pkg/cloudprovider/providers/azure/azure_metrics.go]: 2018

File: pkg/cloudprovider/providers/azure/azure_client.go [hunk 2]
```diff
@@ -84,7 +84,7 @@ type VirtualMachineScaleSetsClient interface {
 	CreateOrUpdate(resourceGroupName string, VMScaleSetName string, parameters compute.VirtualMachineScaleSet, cancel <-chan struct{}) (<-chan compute.VirtualMachineScaleSet, <-chan error)
 	Get(resourceGroupName string, VMScaleSetName string) (result compute.VirtualMachineScaleSet, err error)
 	List(resourceGroupName string) (result compute.VirtualMachineScaleSetListResult, err error)
-	ListNextResults(lastResults compute.VirtualMachineScaleSetListResult) (result compute.VirtualMachineScaleSetListResult, err error)
+	ListNextResults(resourceGroupName string, astResults compute.VirtualMachineScaleSetListResult) (result compute.VirtualMachineScaleSetListResult, err error)
```
Comment (andyzhangx) [inline pkg/cloudprovider/providers/azure/azure_client.go]: lastResults

**Review by feiskyer — APPROVED** [review]
/approve no-issue

**Comment by djsly** [comment]
Quick question, what is the endpoint for having our prometheus stack scrape the cloud-provider metrics ?

Is this done automatically but some kubernetes' internal? Thanks!

**Comment by djsly** [comment]
FYI, I’m stil using the integrated cloud-provider and not the external cloud-provider-manager. So can we access the cloud-provider endpoint when running the integrated version in kubelet ?

**Comment by omerlh** [comment]
@djsly did you find an answer? @ccojocar I just installed 1.10 on Azure and I don't see these metrics, should I add a job?

**Comment by djsly** [comment]
@omerlh , I think you need to use the external cloud provider for access the /metrics, something that I'm not using since the azure cloud provider is still integrated within the upstream code

**Comment by omerlh** [comment]
Ok, I hope someone from sig azure will be able to answer that question...

**Comment by feiskyer** [comment]
@omerlh You could get them from kube-controller-manager's metrics API.

**Comment by omerlh** [comment]
So I should added it the annonation so prometheus will scrape it?

**Comment by omerlh** [comment]
So I need to add an annotation to `kube-controller-manager` in order for prometheus to scrape it?

**Pull request merged**